File: Bwl.Framework/src/Logger/LogWriters/Realizations/FormLogInfo.vb
Public Class FormLogInfo

	Public Property LogInfoText As List(Of String)

	Private Sub _btn_CopyText_Click(sender As Object, e As EventArgs) Handles _btn_CopyText.Click
		Dim buff = String.Empty
		For i = 0 To LogInfoText.Count - 2
			buff += LogInfoText(i) + vbCrLf
		Next
		buff += vbCrLf + LogInfoText(LogInfoText.Count - 1)
		Clipboard.SetText(buff)
	End Sub

	Private Sub _btn_CloseForm_Click(sender As Object, e As EventArgs) Handles _btn_CloseForm.Click
		Close()
	End Sub

	Private Sub FormLogInfo_Shown(sender As Object, e As EventArgs) Handles MyBase.Shown
		_rtb_LogInfo.Text = "["
		For i = 0 To LogInfoText.Count - 2
			_rtb_LogInfo.Text += " " + LogInfoText(i)
		Next
		_rtb_LogInfo.Text += "]" + vbCrLf
		_rtb_LogInfo.Text += LogInfoText(LogInfoText.Count - 1)
	End Sub
End Class

File: Framework/Framework/Bwl.Framework.Windows/Logger/LogWriters/Realizations/FormLogInfo.cs
using System;
using System.Collections.Generic;
using System.Runtime.Versioning;
using System.Windows.Forms;

namespace Bwl.Framework.Windows
{

    [SupportedOSPlatform("windows")]
    public partial class FormLogInfo
    {

        public List<string> LogInfoText { get; set; }

        public FormLogInfo()
        {
            InitializeComponent();
        }
        private void _btn_CopyText_Click(object sender, EventArgs e)
        {
            string buff = string.Empty;
            for (int i = 0, loopTo = LogInfoText.Count - 2; i <= loopTo; i++)
                buff += LogInfoText[i] + Environment.NewLine;
            buff += Environment.NewLine + LogInfoText[LogInfoText.Count - 1];
            Clipboard.SetText(buff);
        }
        private void _btn_CloseForm_Click(object sender, EventArgs e)
        {
            Close();
        }
        private void FormLogInfo_Shown(object sender, EventArgs e)
        {
            _rtb_LogInfo.Text = "[";
            for (int i = 0, loopTo = LogInfoText.Count - 2; i <= loopTo; i++)
                _rtb_LogInfo.Text += " " + LogInfoText[i];
            _rtb_LogInfo.Text += "]" + Environment.NewLine;
            _rtb_LogInfo.Text += LogInfoText[LogInfoText.Count - 1];
        }
    }
}